lib/src/git_backend.rs

```diff
@@ -390,9 +392,17 @@ impl Backend for GitBackend {
             committer,
         };
 
-        let table = self.extra_metadata_store.get_head().map_err(|err| {
-            BackendError::Other(format!("Failed to read non-git metadata: {err}"))
-        })?;
+        let table = {
+            let mut locked_head = self.cached_extra_metadata.lock().unwrap();
+            match locked_head.take() {
```

Comment:
I'm confused about this `.take()`. It seems like it would remove the cached value, leaving a `None` in the mutex, so the next call would have to look it up again. `match locked_head.as_ref()` seems more appropriate.

Reply:
Whoops; you're right: this should've been an `as_ref` with a `.clone` instead. Done.